network/stream/messages.go

```diff
@@ -226,14 +226,23 @@ func (p *Peer) handleOfferedHashesMsg(ctx context.Context, req *OfferedHashesMsg
 	for i := 0; i < lenHashes; i += HashSize {
 		hash := hashes[i : i+HashSize]
 
-		if wait := c.NeedData(ctx, hash); wait != nil {
+		log.Trace("checking offered hash", "ref", fmt.Sprintf("%x", hash))
+
+		if shouldNOTRequestAgain, wait := c.NeedData(ctx, hash); wait != nil {
```

Comment:
@nonsense can you please add a comment here specifically denoting the case that we're handling? this is not all that obvious to me at a first glance

Reply:
@acud I will probably skip this fix, and leave it for a future PR, in order to reduce the scope of this one.

However to answer your question:

When many peers offer the same chunk, without this change, our node might mark it as `wanted` for all peers, when it is enough only for one peer to actually deliver it. So this change is about returning information on whether we want a particular chunk during syncing, and whether we have to request it.

Reply:
@acud I have disabled this, so that we reduce the scope of the PR and have a more iterative approach.

Reply:
thanks :+1: